src/pgduckdb_hooks.cpp

```diff
@@ -46,20 +46,18 @@ IsAllowedStatement(Query *query) {
 	if (query->hasModifyingCTE) {
 		return false;
 	}
-	/* For `SELECT ..` ActivePortal doesn't exist */
-	if (!ActivePortal)
-		return true;
-	/* `EXPLAIN ...` should be allowed */
-	if (ActivePortal->commandTag == CMDTAG_EXPLAIN)
-		return true;
-	return false;
+	/* We don't support modifying statements yet */
+	if (query->commandType != CMD_SELECT) {
```

Comment:
We already check this condition in `DuckdbPlannerHook` if statement but this is better place to contain this check `commandType == SELECT`.

Reply:
I moved a few more checks from the DuckDBPlannerHook to IsAllowedStatement now.